AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLModelBasedTests+Subscriptions.swift

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AWSMobileClient
+@testable import AWSAPICategoryPlugin
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSAPICategoryPluginTestCommon
+@testable import AppSyncSubscriptionClient
+
+/// Subscription related stress/load tests
+extension GraphQLModelBasedTests {
+
+    /// Ensure that the underlying websocket socket is disconnected when the number of subscriptions on the socket
+    /// reaches zero. This test first creates multiple subscriptions, does a mutation to make sure data is flowing,
+    /// cancel the operations, and makes sure the underlying socket is disconnected. Given this new state of the system
+    /// where there is an existing connection provider, run a similar test by create a new subscription, which in turn
```

Comment:
You might want to rephrase this paragraph (feel free to reject/accept, and/or update where is necessary)

The purpose of this test is to ensure that all websockets can be successfully created, exercised and terminated while keeping a single connection provider in memory.  Specifically, the following test exercises the following:
1.  Create a new connection provider
2.  Create multiple subscriptions
3.  Make a mutation and validate that it succeeds
4.  Cancel the operations
5.  Ensure the socket is disconnected
6.  Ensure connection provider is still in memory
7.  Repeat Steps 2-6 with the existing connection provider.

Reply:
thanks for the improved wording! will take this with next iteration